pkg/reconciler/taskrun/taskrun.go

```diff
@@ -182,38 +179,60 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return multierror.Append(err, c.updateStatusLabelsAndAnnotations(tr, original)).ErrorOrNil()
 	}
 
+	// prepare fetches all required resources, validates them together with the
```

Comment:
suggest this comment describing `prepare`'s behaviour be attached to the `func prepare(...)` on line 214.  But I do think it makes sense to leave a comment here about errors coming back from prepare being considered permanent.

Reply:
Good point, will do